File: 2073/solution.ts
function timeRequiredToBuy(tickets: number[], k: number): number {
  let res = 0;
  for (let i = 0; i < tickets.length; i++) {
    if (i == k) {
      res += tickets[k];
    }
    if (i < k) {
      res += Math.min(tickets[i], tickets[k]);
    }

    if (i > k) {
      res += Math.min(tickets[i], tickets[k] - 1);
    }
  }
  return res;
}

File: 2073/solution.go
package main

func timeRequiredToBuy(tickets []int, k int) int {
	res := 0
	for i, v := range tickets {
		if i < k {
			res += min(v, tickets[k])
		} else if i == k {
			res += v
		} else {
			res += min(v, tickets[k]-1)
		}
	}
	return res
}

func min(a int, b int) int {
	if a < b {
		return a
	}
	return b
}